### PR TITLE
 HHH-13141 use upper-case for sequences query in SQLServer2012Dialect

### PR DESCRIPTION
**Pull request by dohertyfjatl**
The upper-case query should work on both case-sensitive and case-insensitive collations.

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/test/dialect/functional/SQLServerDialectSequenceInformationTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+				log.error( e );
+				fail( "Sequence information was not retrieved: " + e.getMessage() );
+			}
+			finally {
+				StandardServiceRegistryBuilder.destroy( ssr );
+			}
+		} );
+	}
+
+	private void doInAutoCommit( String updateQuery ) {
```
Comment (vladmihalcea) [inline hibernate-core/src/test/java/org/hibernate/test/dialect/functional/SQLServerDialectSequenceInformationTest.java]: We have a `doInAutoCommit``method in `AbstractStoredProcedureTest`. I think we should move that to `TransactionUtil`.

File: hibernate-core/src/test/java/org/hibernate/test/dialect/functional/SQLServerDialectSequenceInformationTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	@TestForIssue(jiraKey = "HHH-13141")
+	public void testExtractSequenceInformationForSqlServerWithCaseSensitiveCollation() {
+		String databaseNameToken = "databaseName=";
+		String url = (String) Environment.getProperties().get( AvailableSettings.URL );
+		String[] tokens = url.split( databaseNameToken );
+		String newUrl = tokens[0] + databaseNameToken + DATABASE_NAME;
+
+		Dialect dialect = serviceRegistry().getService( JdbcServices.class ).getDialect();
+
+		doInHibernate( this::sessionFactory, session -> {
```
Comment (vladmihalcea) [inline hibernate-core/src/test/java/org/hibernate/test/dialect/functional/SQLServerDialectSequenceInformationTest.java]: The call to `doInHibernate` is not needed. You can just remove it and execute its content instead.

**Comment by gsmet** [comment]
@vladmihalcea I let you take care of this issue since you started reviewing it?

My only comment is that we need to squash the whitespace fix commits.

This one is needed for 5.4.0.Final I plan to release next week.

**Comment by vladmihalcea** [comment]
@gsmet Sure thing. I have some changes I think we should do to this issue like verifying we extract the sequence info for a given sequence.

**Comment by gsmet** [comment]
Superseded by https://github.com/hibernate/hibernate-orm/pull/2678 .

**Pull request closed without merging**